pkg/controllers/clusterresourceplacementeviction/controller.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package clusterresourceplacementeviction
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/klog/v2"
+	runtime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/controller"
+
+	placementv1alpha1 "go.goms.io/fleet/apis/placement/v1alpha1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/utils/condition"
+	"go.goms.io/fleet/pkg/utils/controller"
+)
+
+const (
+	reasonClusterResourcePlacementEvictionValid       = "ClusterResourcePlacementEvictionValid"
+	reasonClusterResourcePlacementEvictionInvalid     = "ClusterResourcePlacementEvictionInvalid"
+	reasonClusterResourcePlacementEvictionExecuted    = "ClusterResourcePlacementEvictionExecuted"
+	reasonClusterResourcePlacementEvictionNotExecuted = "ClusterResourcePlacementEvictionNotExecuted"
+
+	evictionInvalidMissingCRP  = "Failed to find cluster resource placement targeted by eviction"
```

Comment:
Hi Arvind! Just a nit: would you mind adding a `Message` suffix/prefix to the variables, just to make things a bit more clear?

Reply:
Also, binding are generally speaking considered to be our internal APIs, to keep things a bit more clear on the user end, it might make more sense to just use cluster references in the message part.